part2/stages/Functions/install-main

```diff
@@ -872,16 +872,27 @@ install_file()
     local SRC="$1"
     local DST="$2"
     local PACKAGE_TYPE="$3"
+    local config_mounted="0"
+    local RET="0"
 
-    mkdir -p $(dirname "${DOM0_MOUNT}/${DST}") || return 1
+    if [ "$( expr substr "$DST" 1 8 )" = "/config/" ] ; then
+        config_mounted="1"
+        mount_config
+    fi
+
+    mkdir -p $(dirname "${DOM0_MOUNT}/${DST}") || RET="1"
     rm -f "${DOM0_MOUNT}/${DST}.new" 2>/dev/null # ignore errors on this command.
     if [ "${PACKAGE_TYPE}" = "gz" ] ; then
-        do_cmd gunzip -c "${SRC}" > "${DOM0_MOUNT}/${DST}.new" || return 1
+        do_cmd gunzip -c "${SRC}" > "${DOM0_MOUNT}/${DST}.new" || RET="1"
     else
-        cp "${SRC}" "${DOM0_MOUNT}/${DST}.new" || return 1
+        cp "${SRC}" "${DOM0_MOUNT}/${DST}.new" || RET="1"
     fi
 
-    return 0
+    if [ "$config_mounted" = "1" ] ; then
```

Comment:
Actually, the existing checks we have for ```is_mounted ${DOM0_MOUNT}/config``` like on line 419 is probably sufficient instead of this flag.

Reply:
The idea of config_mounted is that we should only un-mount if the mount was performed here.  As I discovered, a fresh install leaves "${DOM0_MOUNT}/config" mounted, and this would un-mount it, potentially even if we aren't installing into /config.  So it's a bit of a mess, but I think it's "safer" to keep it as-is.